webapi/Services/AzureFormRecognizerOcrEngine.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Azure;
+using Azure.AI.FormRecognizer;
+using Azure.AI.FormRecognizer.Models;
+using Microsoft.AspNetCore.Http;
+
+namespace SemanticKernel.Service.Services;
+
+/// <summary>
+/// Wrapper for the Azure.AI.FormRecognizer. This allows Form Recognizer to be used as the OCR engine for reading text from files with an image MIME type.
+/// </summary>
+public class AzureFormRecognizerOcrEngine : IOcrEngine
+{
+    /// <summary>
+    /// Creates a new instance of the AzureFormRecognizerOcrEngine passing in the Form Recognizer endpoint and key.
+    /// </summary>
+    /// <param name="endpoint">The endpoint for accessing a provisioned Azure Form Recognizer instance</param>
+    /// <param name="credential">The AzureKeyCredential containing the provisioned Azure Form Recognizer access key</param>
+    public AzureFormRecognizerOcrEngine(string endpoint, AzureKeyCredential credential)
+    {
+        this.FormRecognizerClient = new FormRecognizerClient(new Uri(endpoint), credential);
+    }
+
+    public FormRecognizerClient FormRecognizerClient { get; }
+
+    ///<inheritdoc/>
+    public async Task<string> ReadTextFromImageFileAsync(IFormFile imageFile)
+    {
+        await using (var ms = new MemoryStream())
+        {
+            await imageFile.CopyToAsync(ms);
+            var fileBytes = ms.ToArray();
+            await using var imgStream = new MemoryStream(fileBytes);
```

Comment:
I'm curious. Why do we need to create a second memory stream?

Reply:
We don't!

Reply:
Isn't there one on line 34 and one on line 38?

Could we not just do

```
await using (var imgStream = new MemoryStream())
{
    await imageFile.CopyToAsync(imgStream);
    ...
```

Reply:
Yes, there were 2, the PR should've been updated yesterday to only include one.   This is the link to updated file here - https://github.com/microsoft/chat-copilot/pull/26/files#diff-d7d3889b894f3d930178908a909f051b7caf23ea8302fc08e63f06689b0677aa

Reply:
Yes, saw that now.  Closing this convo.  But there's another similar piece of code in the Tesseract engine wrapper...